test/governance/candidacy.users.test.ts

```diff
@@ -0,0 +1,175 @@
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { getEnvironmentRequiredVars, waitForNBlocks } from "../../utils/utils";
+
+const { chainUri: environmentUri } = getEnvironmentRequiredVars();
+const termDuration = 50; // Set as a const in mangata-node
+
+let bootnode: Node;
+let keyring: Keyring;
+let sudo: SudoUser;
+
+let candidate: GovernanceUser;
+let voter: GovernanceUser;
+let bank: Bank;
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(15000000);
+
+beforeAll(async () => {
+  await cryptoWaitReady(); // Wait for Polkadots WASM backend
+
+  bootnode = new Node(environmentUri);
+  await bootnode.connect();
+  await bootnode.subscribeToHead();
+
+  keyring = new Keyring({ type: "sr25519" });
+  sudo = UserFactory.createUser(Users.SudoUser, keyring, bootnode) as SudoUser;
+});
+
+beforeEach(async () => {
+  // Don't run tests at the end of a term
+  if (termDuration % bootnode.lastBlock! >= 45) {
+    waitForNextTerm();
+  }
+
+  // Initialize actors
+  candidate = UserFactory.createUser(
+    Users.GovernanceUser,
+    keyring,
+    bootnode
+  ) as GovernanceUser;
+  voter = UserFactory.createUser(
+    Users.GovernanceUser,
+    keyring,
+    bootnode
+  ) as GovernanceUser;
+  bank = new Bank(sudo);
+
+  // Fund accounts
+  await voter.addMGATokens(bank.sudoUser, new BN(Math.pow(10, 17).toString()));
+  await candidate.addMGATokens(
+    bank.sudoUser,
+    new BN(Math.pow(10, 16).toString())
+  );
+
+  // Subscribe to events
+  candidate.node.subscribeToUserBalanceChanges(candidate);
+  voter.node.subscribeToUserBalanceChanges(voter);
+});
+
+describe("Governance -> Candidacy -> Users", () => {
+  test("User can run for candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    await candidate.runForCouncil();
+    await waitForNBlocks(2);
+
+    const candidates = getLastBlocksElectionData();
+
+    expect(candidates).toEqual(expect.arrayContaining([candidatesAddress]));
+  });
+
+  test("Candidate can renounce candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    await candidate.runForCouncil();
+    await waitForNBlocks(2);
+    await candidate.renounceCandidacy();
+    await waitForNBlocks(2);
+
+    const candidates = getLastBlocksElectionData();
+
+    expect(candidates).not.toEqual(expect.arrayContaining([candidatesAddress]));
+  });
+
+  test("Council Member can renounce candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    // Gonzalo runs for Council in Term 0
+    // Eddy votes for Gonzalo
+    await candidate.runForCouncil();
+    await voter.vote([candidate], new BN(Math.pow(10, 16).toString()));
+    await waitForNBlocks(2);
+
+    const candidatesTerm0 = getLastBlocksElectionData();
+
+    expect(candidatesTerm0).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    // Gonzalo wins the election and is council member!
```

Comment:
Should not we validate that Gonzalo is in the council?

Reply:
Yeah you're right